File: ConstructionWorkers/src/ASTnodes/statements/Statement.java
/**
 * Statement.java.
 */

package ASTnodes.statements;

import ASTnodes.LineNumber;
import ASTnodes.Node;
import ASTnodes.visitors.FormAndStatementVisitor;

public abstract class Statement extends Node {

    public Statement(LineNumber location) {
        super(location);
    }

    public abstract <T> T accept(FormAndStatementVisitor<T> visitor);
}

File: OffByOne/src/OffByOne.Ql/Ast/Statements/Base/Statement.cs
namespace OffByOne.Ql.Ast.Statements.Base
{
    using System.Collections.Generic;

    using OffByOne.Ql.Common.Visitors.Contracts;

    public abstract class Statement : AstNode, IVisitableStatement
    {
        public abstract TResult Accept<TResult, TContext>(IStatementVisitor<TResult, TContext> visitor, TContext environment)
            where TContext : IEnvironment;

        public abstract ISet<string> GetDependencies();
    }
}
